packages/uicore/src/components/ScrollContainer/ScrollContainer.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { CSSProperties, FC, useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react'
+import cx from 'classnames'
+import rafSchd from 'raf-schd'
+import css from './ScrollContainer.css'
+
+export interface ScrollContainerProps {
+  /**
+   * `true`: renders top shadow when required.
+   *
+   * `false`: never renders top shadow.
+   *
+   * default: `true` */
+  showTopShadow?: boolean
+  /**
+   * `true`: renders bottom shadow when required
+   *
+   * `false`: never renders bottom shadow
+   *
+   * default: `true` */
+  showBottomShadow?: boolean
+  /**
+   * height of the container
+   *
+   * default: `100%` */
+  height?: CSSProperties['height']
+  containerClassName?: string
```

Comment:
The `container` part is probably a little overly-specific. As a consumer, I would expect `className` like other components.